diff --git a/compiler/cgbase.pas b/compiler/cgbase.pas
index 7b23da08d8..2c3aba12aa 100644
--- a/compiler/cgbase.pas
+++ b/compiler/cgbase.pas
@@ -93,16 +93,19 @@ interface
          addr_low_call,    // counterpart of two above, generate call_hi16 and call_lo16 relocs
          addr_high_call
          {$ENDIF}
-         {$if defined(RISCV32) or defined(RISCV64)}
+         {$if defined(RISCV32) or defined(RISCV64) or defined(SLOW32)}
          ,
          addr_hi20,
-         addr_lo12,
+         addr_lo12
+         {$if defined(RISCV32) or defined(RISCV64)}
+         ,
          addr_pcrel_hi20,
          addr_pcrel_lo12,
          addr_pcrel,
          addr_got_pcrel_hi,
          addr_plt
          {$endif RISCV}
+         {$endif}
          {$if defined(LOONGARCH64)}
          ,
          addr_b16, { %b16(sym) }
diff --git a/compiler/fpcdefs.inc b/compiler/fpcdefs.inc
index 51d296ae62..d1a831943f 100644
--- a/compiler/fpcdefs.inc
+++ b/compiler/fpcdefs.inc
@@ -416,6 +416,20 @@
   {$define support_rodata}
 {$endif loongarch64}
 
+{$ifdef slow32}
+  {$define cpu32bit}
+  {$define cpu32bitaddr}
+  {$define cpu32bitalu}
+  {$define cputargethasfixedstack}
+  {$define cpurequiresproperalignment}
+  {$define cpufpemu}
+  {$define cpuneedsmulhelper}
+  {$define cpuneedsdivhelper}
+  {$define cpucapabilities}
+  {$define support_rodata}
+  {$define SUPPORT_SAFECALL}
+{$endif slow32}
+
 { Stabs is not officially supported on 64 bit targets by gdb, except on Mac OS X
   (but there we don't support it)
 }
diff --git a/compiler/globals.pas b/compiler/globals.pas
index 0c5bdbc7ea..3ddde07c6f 100644
--- a/compiler/globals.pas
+++ b/compiler/globals.pas
@@ -676,6 +676,12 @@       tpendingstate = record
         asmcputype : cpu_none;
         fputype : fpu_fd;
   {$endif loongarch64}
+  {$ifdef slow32}
+        cputype : cpuinfo.cpu_slow32;
+        optimizecputype : cpuinfo.cpu_slow32;
+        asmcputype : cpu_none;
+        fputype : fpu_soft;
+  {$endif slow32}
 {$endif not GENERIC_CPU}
         asmmode : asmmode_standard;
 {$ifndef jvm}
diff --git a/compiler/options.pas b/compiler/options.pas
index 1b5203c70a..f109c349ed 100644
--- a/compiler/options.pas
+++ b/compiler/options.pas
@@ -4832,6 +4832,14 @@ procedure read_arguments(cmd:TCmdStr);
         def_system_macro('FPC_LOCALS_ARE_STACK_REG_RELATIVE');
       {$endif loongarch64}
 
+      {$ifdef slow32}
+        def_system_macro('CPUSLOW32');
+        def_system_macro('CPU32');
+        def_system_macro('FPC_CURRENCY_IS_INT64');
+        def_system_macro('FPC_COMP_IS_INT64');
+        def_system_macro('FPC_REQUIRES_PROPER_ALIGNMENT');
+      {$endif slow32}
+
       {$if defined(cpu8bitalu)}
         def_system_macro('CPUINT8');
       {$elseif defined(cpu16bitalu)}
@@ -5352,7 +5360,7 @@ procedure read_arguments(cmd:TCmdStr);
          system_arm_nds,system_arm_embedded,system_arm_freertos,
          system_riscv32_embedded,system_riscv64_embedded,system_xtensa_linux,
          system_z80_embedded,system_z80_zxspectrum,system_riscv32_freertos,
-         system_mipsel_ps1])
+         system_mipsel_ps1,system_slow32_embedded])
 {$ifdef arm}
       or (target_info.abi=abi_eabi)
 {$endif arm}
diff --git a/compiler/version.pas b/compiler/version.pas
index 1b3ca1c5ae..b883fa0397 100644
--- a/compiler/version.pas
+++ b/compiler/version.pas
@@ -89,6 +89,9 @@ interface
 {$ifdef cpuloongarch64}
         source_cpu_string = 'loongarch64';
 {$endif cpuloongarch64}
+{$ifdef cpuslow32}
+        source_cpu_string = 'slow32';
+{$endif cpuslow32}
 
 function version_string:string;
 function full_version_string:string;
