diff --git a/hw/Kconfig b/hw/Kconfig
index c92ca2b13a..144229e689 100644
--- a/hw/Kconfig
+++ b/hw/Kconfig
@@ -61,6 +61,7 @@ source m68k/Kconfig
 source microblaze/Kconfig
 source mips/Kconfig
 source or1k/Kconfig
+source slow32/Kconfig
 source ppc/Kconfig
 source riscv/Kconfig
 source rx/Kconfig
diff --git a/hw/meson.build b/hw/meson.build
index 7fa81db453..d69f80dc1f 100644
--- a/hw/meson.build
+++ b/hw/meson.build
@@ -11,6 +11,7 @@ subdir('m68k')
 subdir('microblaze')
 subdir('mips')
 subdir('or1k')
+subdir('slow32')
 subdir('ppc')
 subdir('remote')
 subdir('riscv')
