diff --git a/hw/Kconfig b/hw/Kconfig
index 9e6c789ae7..2f228c9576 100644
--- a/hw/Kconfig
+++ b/hw/Kconfig
@@ -58,7 +58,8 @@ source loongarch/Kconfig
 source m68k/Kconfig
 source microblaze/Kconfig
 source mips/Kconfig
-source openrisc/Kconfig
+source or1k/Kconfig
+source slow32/Kconfig
 source ppc/Kconfig
 source riscv/Kconfig
 source rx/Kconfig
diff --git a/hw/meson.build b/hw/meson.build
index 1022bdb806..bb240dce46 100644
--- a/hw/meson.build
+++ b/hw/meson.build
@@ -9,7 +9,8 @@ subdir('loongarch')
 subdir('m68k')
 subdir('microblaze')
 subdir('mips')
-subdir('openrisc')
+subdir('or1k')
+subdir('slow32')
 subdir('ppc')
 subdir('remote')
 subdir('riscv')
