diff --git a/hw/Kconfig b/hw/Kconfig
index f8f92b5d03..2f228c9576 100644
--- a/hw/Kconfig
+++ b/hw/Kconfig
@@ -59,6 +59,7 @@ source m68k/Kconfig
 source microblaze/Kconfig
 source mips/Kconfig
 source or1k/Kconfig
+source slow32/Kconfig
 source ppc/Kconfig
 source riscv/Kconfig
 source rx/Kconfig
diff --git a/hw/meson.build b/hw/meson.build
index 66e46b8090..bb240dce46 100644
--- a/hw/meson.build
+++ b/hw/meson.build
@@ -10,6 +10,7 @@ subdir('m68k')
 subdir('microblaze')
 subdir('mips')
 subdir('or1k')
+subdir('slow32')
 subdir('ppc')
 subdir('remote')
 subdir('riscv')
